### ISSUE TITLE
Allow the scheduler to be None

### ISSUE THREAD

**Issue by BramVanroy** [PyTorchLightning/pytorch-lightning#1641]
## 🚀 Feature

Allow a scheduler to be None. This means that one could have `configure_optimizers()` return `[optimizer], [None]`.

### Motivation

Allowing `configure_optimizers()` to return `[optimizer], [None]` improves how one could write clean, dynamic code. One could for instance do something like this:

```python
def configure_optimizers(self):
    self.optim = create_optimizer(self.model, **self.optim_opts)
    self.lr_scheduler = create_lr_scheduler(self.optim, **self.sched_opts)
    return [self.optim], [self.lr_scheduler]
```
where `create_lr_scheduler()` can return None when no scheduler is needed. 

### Alternatives

I am aware that one could rewrite as follows, but this is a lot less clean and can easily be changed in the source code.

```python
def configure_optimizers(self):
    self.optim = create_optimizer(self.model, **self.optim_opts)
    self.lr_scheduler = create_lr_scheduler(self.optim, **self.sched_opts)

    if self.lr_scheduler is None:
        return self.optim
    else:
        return [self.optim], [self.lr_scheduler]
```

### Additional context

I can do a PR for this. It should be as easy as adding a condition in configure_schedulers()

https://github.com/PyTorchLightning/pytorch-lightning/blob/e79ae18cae352505524215b6f3617f862ad024f2/pytorch_lightning/trainer/optimizers.py#L83

namely, allowing:
```python
elif scheduler is None:
     continue
```

**Comment by BramVanroy** [comment]
Bump.

**Comment by chiragraman** [comment]
Found this while checking if anyone else had faced this, @BramVanroy did this go anywhere? Seems like I'm 5 months late to the party.

**Comment by BramVanroy** [comment]
@chiragraman I haven't heard anything so I don't think this has been changed.

**Comment by chiragraman** [comment]
@BramVanroy So it seems that this works, which maybe a reasonable compromise to avoid multiple return paths:

```
def configure_optimizers(self):
    """ Setup optimizers for training the Social Process """
    optimizer = Adam(
        list(self.process.parameters()) + list(self.loss.parameters()),
        lr=self.hparams.lr, weight_decay=self.hparams.weight_decay
    )
    scheduler = (lr_scheduler.MultiStepLR(optimizer, milestones=self.hparams.lr_steps)
                 if self.hparams.lr_sched else None)
    return {"optimizer": optimizer, "lr_schedulers": scheduler}
```
